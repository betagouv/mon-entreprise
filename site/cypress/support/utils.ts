export const fr = Cypress.env('language') === 'fr'
export const baseUrl = Cypress.config('baseUrl')
