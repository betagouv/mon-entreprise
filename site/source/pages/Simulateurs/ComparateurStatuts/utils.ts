export type ValueType =
	| string
	| number
	| boolean
	| null
	| Record<string, unknown>
export type BestOption = {
	type: 'sasu' | 'ei' | 'ae'
	value?: ValueType
}
export const getBestOption = (options: BestOption[]) => {
	const sortedOptions = options.sort(
		(option1: BestOption, option2: BestOption) => {
			if (option1.value === null || option1.value === undefined) {
				return 1
			}
			if (option2.value === null || option2.value === undefined) {
				return -1
			}

			if (option1.value === option2.value) return 0
			// console.log(option1.value, option2.value, option1.value > option2.value)

			return option1.value > option2.value ? -1 : 1
		}
	)

	return sortedOptions?.[0]?.type
}
