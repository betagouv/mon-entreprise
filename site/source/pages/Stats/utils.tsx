import { Strong } from '@/design-system/typography'
import { H4 } from '@/design-system/typography/heading'
import { Intro, SmallBody } from '@/design-system/typography/paragraphs'
import React from 'react'
import styled from 'styled-components'

export type IndicatorProps = {
	main?: React.ReactNode
	subTitle?: React.ReactNode
	footnote?: React.ReactNode
}
export function Indicator({ main, subTitle, footnote }: IndicatorProps) {
	return (
		<StyledIndicator>
			<H4 as="h2">{subTitle}</H4>
			<Intro>
				<Strong>{main}</Strong>
			</Intro>
			{footnote && (
				<SmallBody
					css={`
						margin-top: -1rem;
					`}
				>
					{footnote}
				</SmallBody>
			)}
		</StyledIndicator>
	)
}
export function formatDay(date: string | Date) {
	return new Date(date).toLocaleString('default', {
		weekday: 'long',
		day: 'numeric',
		month: 'long',
	})
}
export function formatMonth(date: string | Date) {
	return new Date(date).toLocaleString('default', {
		month: 'long',
		year: 'numeric',
	})
}

const StyledIndicator = styled.div`
	border-radius: ${({ theme }) => theme.box.borderRadius};
	height: 100%;
	box-shadow: ${({ theme }) => theme.elevations[2]};
	padding: ${({ theme }) => theme.spacings.xs}
		${({ theme }) => theme.spacings.md};
`
