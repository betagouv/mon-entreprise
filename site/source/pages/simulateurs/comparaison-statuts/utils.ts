import { DottedName } from 'modele-social'
import Engine, { Evaluation } from 'publicodes'

import { StatutType } from '@/components/StatutTag'

export type OptionType = {
	engine: Engine<DottedName>
	name: StatutType
	value: Evaluation
}
export const getBestOption = (options: OptionType[], leastIsBest: boolean) => {
	const sortedOptions = [...options].sort(
		(option1: OptionType, option2: OptionType) => {
			if (option1.value === null || option1.value === undefined) {
				return leastIsBest ? -1 : 1
			}
			if (option2.value === null || option2.value === undefined) {
				return leastIsBest ? 1 : -1
			}

			if (option1.value === option2.value) return 0

			return (option1.value > option2.value ? -1 : 1) * (leastIsBest ? -1 : 1)
		}
	)

	return sortedOptions?.[0]?.name
}
