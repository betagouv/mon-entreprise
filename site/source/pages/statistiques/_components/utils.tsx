export function formatDay(date: string | Date, language: string) {
	return new Date(date).toLocaleString(language, {
		weekday: 'long',
		day: 'numeric',
		month: 'long',
	})
}
export function formatMonth(date: string | Date, language: string) {
	return new Date(date).toLocaleString(language, {
		month: 'long',
		year: 'numeric',
	})
}
export function formatIndicator(number: number, language: string) {
	return Intl.NumberFormat(language, {
		notation: 'compact',
	}).format(number)
}

export function formatProgression(
	previousValue: number,
	currentValue: number,
	language: string
) {
	return Intl.NumberFormat(language, {
		style: 'percent',
		signDisplay: 'exceptZero',
	}).format(currentValue / previousValue - 1)
}
export function emojiSatisfaction(satisfaction: number) {
	return satisfaction > 0.85
		? '😀'
		: satisfaction > 0.65
		? '🙂'
		: satisfaction > 0.5
		? '😐'
		: '😕'
}

export function messageTypeSatisfaction(satisfaction: number) {
	return satisfaction > 0.85
		? 'success'
		: satisfaction > 0.65
		? 'primary'
		: satisfaction > 0.5
		? 'info'
		: 'error'
}
