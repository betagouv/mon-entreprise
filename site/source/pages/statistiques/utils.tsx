import React from 'react'
import { styled } from 'styled-components'

import { Strong } from '@/design-system/typography'
import { H4 } from '@/design-system/typography/heading'
import { Intro, SmallBody } from '@/design-system/typography/paragraphs'

export type IndicatorProps = {
	main?: React.ReactNode
	subTitle?: React.ReactNode
	footnote?: React.ReactNode
}
export function Indicator({ main, subTitle, footnote }: IndicatorProps) {
	return (
		<StyledIndicator>
			<H4 as="h2">{subTitle}</H4>
			<Intro as="div">
				<Strong>{main}</Strong>
			</Intro>
			{footnote && <SmallBody>{footnote}</SmallBody>}
		</StyledIndicator>
	)
}
export function formatDay(date: string | Date) {
	return new Date(date).toLocaleString('default', {
		weekday: 'long',
		day: 'numeric',
		month: 'long',
	})
}
export function formatMonth(date: string | Date) {
	return new Date(date).toLocaleString('default', {
		month: 'long',
		year: 'numeric',
	})
}

const StyledIndicator = styled.div`
	border-radius: ${({ theme }) => theme.box.borderRadius};
	height: 100%;
	box-shadow: ${({ theme }) =>
		theme.darkMode ? theme.elevationsDarkMode[2] : theme.elevations[2]};
	padding: ${({ theme }) => theme.spacings.xs}
		${({ theme }) => theme.spacings.md};
`
