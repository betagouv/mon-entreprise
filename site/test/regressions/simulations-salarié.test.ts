import { expect, it } from 'vitest'
import employeeConfig from '../../source/pages/Simulateurs/configs/salarié.yaml'
import employeeSituations from './simulations-salarié.yaml'
import { engine, getMissingVariables, runSimulations } from './utils'

it('calculate simulations-salarié', () => {
	runSimulations(
		employeeSituations,
		employeeConfig.objectifs,
		employeeConfig.situation
	)

	expect(
		getMissingVariables(
			engine
				.setSituation({
					...employeeConfig.situation,
					'salarié . contrat . salaire brut': '3000 €/mois',
				})
				.evaluate('salarié . rémunération . net après impôt')
		)
	).toMatchInlineSnapshot(`
		[
		  "impôt . méthode de calcul",
		  "salarié . complémentaire santé . forfait",
		  "salarié . complémentaire santé . part employeur",
		  "salarié . contrat",
		  "salarié . contrat . temps de travail . temps partiel",
		  "salarié . convention collective",
		  "salarié . déduction forfaitaire spécifique",
		  "salarié . régime des impatriés",
		  "salarié . rémunération . avantages en nature",
		  "salarié . rémunération . frais professionnels . abonnement transports publics . montant",
		  "salarié . rémunération . frais professionnels . titres-restaurant",
		  "salarié . rémunération . frais professionnels . transports personnels . carburant faible émission . montant",
		  "salarié . rémunération . frais professionnels . transports personnels . forfait mobilités durables . montant",
		  "salarié . rémunération . primes . activité . base",
		  "salarié . rémunération . primes . fin d'année",
		  "salarié . statut cadre",
		  "salarié . temps de travail . heures supplémentaires",
		  "situation personnelle . domiciliation fiscale à l'étranger",
		  "établissement . localisation",
		]
	`)
})
