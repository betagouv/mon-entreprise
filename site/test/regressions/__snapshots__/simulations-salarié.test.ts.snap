// Vitest Snapshot v1

exports[`calculate simulations-salarié > ATMP 1`] = `"[2370,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > ATMP 2`] = `"[2312,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > ATMP 3`] = `"[2326,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > CCN HCR 1`] = `
"[3480,0,2500,1983,1872]
Notifications affichées : salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > CCN batiment 1`] = `
"[3837,0,2500,1931,1822]
Notifications affichées : salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > CCN batiment 2`] = `
"[3819,0,2500,1933,1824]
Notifications affichées : salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > CCN batiment 3`] = `
"[3869,0,2500,1949,1837]
Notifications affichées : salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > CCN compta 1`] = `
"[3365,0,2500,2006,1898]
Notifications affichées : salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > CCN optique 1`] = `
"[3554,0,2500,2058,1943]
Notifications affichées : salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > CCN spectacle vivant 1`] = `
"[3795,0,2500,2155,2039]
Notifications affichées : salarié . contrat . CDD . information, salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > CCN sport 1`] = `
"[535,0,500,374,374]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > JEI 1`] = `"[3443,0,3000,2353,2168]"`;

exports[`calculate simulations-salarié > JEI 2`] = `
"[26525,0,20000,15997,10736]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > activité partielle 1`] = `
"[58,0,1560,1307,1307]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > activité partielle 2`] = `"[980,0,4000,2242,2118]"`;

exports[`calculate simulations-salarié > activité partielle 3`] = `"[2100,0,8000,4484,3737]"`;

exports[`calculate simulations-salarié > activité partielle 4`] = `"[1614,0,4000,2406,2218]"`;

exports[`calculate simulations-salarié > activité partielle 5`] = `"[2838,0,4000,2686,2409]"`;

exports[`calculate simulations-salarié > activité partielle 6`] = `"[740,3750,3000,1682,1631]"`;

exports[`calculate simulations-salarié > activité partielle 7`] = `"[980,0,4000,2242,2206]"`;

exports[`calculate simulations-salarié > aides 1`] = `"[2302,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > aides 2`] = `
"[12818,0,10000,8925,7687]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > aides 3`] = `"[1521,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > aides 4`] = `
"[2336,0,2000,1720,1667]
Notifications affichées : salarié . contrat . CDD . information"
`;

exports[`calculate simulations-salarié > apprentissage 1`] = `
"[1547,0,1500,1460,1460]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > apprentissage 2`] = `
"[1380,0,1500,1460,1460]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > atmp 1`] = `"[2410,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > avantages 1`] = `"[2542,0,2000,1540,1490]"`;

exports[`calculate simulations-salarié > avantages 2`] = `"[2553,0,2000,1539,1489]"`;

exports[`calculate simulations-salarié > avantages 3`] = `"[2465,0,2000,1549,1514]"`;

exports[`calculate simulations-salarié > cadre 1`] = `"[4116,0,3000,2348,2160]"`;

exports[`calculate simulations-salarié > cadre 2`] = `"[11561,0,8000,6250,4973]"`;

exports[`calculate simulations-salarié > cdd 1`] = `
"[3169,0,2200,1894,1815]
Notifications affichées : salarié . contrat . CDD . information"
`;

exports[`calculate simulations-salarié > cdd 2`] = `
"[3444,0,2200,2009,1897]
Notifications affichées : salarié . contrat . CDD . information"
`;

exports[`calculate simulations-salarié > cdd 3`] = `
"[3734,0,2200,2197,2084]
Notifications affichées : salarié . contrat . CDD . information"
`;

exports[`calculate simulations-salarié > cdd 4`] = `
"[4276,0,2200,2292,2108]
Notifications affichées : salarié . contrat . CDD . information"
`;

exports[`calculate simulations-salarié > cdd 5`] = `
"[3689,0,2200,1871,1792]
Notifications affichées : salarié . contrat . CDD . information, salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > contrat pro 1`] = `"[1566,0,1500,1165,1165]"`;

exports[`calculate simulations-salarié > contrat pro 2`] = `"[1566,0,1500,1165,1165]"`;

exports[`calculate simulations-salarié > effectif 1`] = `"[2354,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > effectif 2`] = `"[2365,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > effectif 3`] = `"[2377,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > effectif 4`] = `"[2377,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > frais pro - DFS 1`] = `"[2078,0,2000,1622,1622]"`;

exports[`calculate simulations-salarié > frais pro - DFS 2`] = `"[2178,0,2000,1584,1549]"`;

exports[`calculate simulations-salarié > frais pro - DFS 3`] = `"[2132,0,2000,1598,1562]"`;

exports[`calculate simulations-salarié > frais pro - DFS 4`] = `"[2132,0,2000,1598,1562]"`;

exports[`calculate simulations-salarié > frais pro - DFS 5`] = `"[2354,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > frais pro - DFS 6`] = `"[1765,0,1700,1351,1351]"`;

exports[`calculate simulations-salarié > frais pro - DFS 7`] = `"[3221,0,2600,2125,2105]"`;

exports[`calculate simulations-salarié > frais pro - abo transports + transports personnels 1`] = `"[4373,0,3200,2545,2348]"`;

exports[`calculate simulations-salarié > frais pro - abo transports + transports personnels 2`] = `"[4896,0,3200,2856,2566]"`;

exports[`calculate simulations-salarié > frais pro - abo transports + transports personnels 3`] = `"[4665,0,3200,2823,2624]"`;

exports[`calculate simulations-salarié > frais pro - abonnement transports publics 1`] = `"[4380,0,3200,2551,2354]"`;

exports[`calculate simulations-salarié > frais pro - abonnement transports publics 2`] = `"[5090,0,3200,3261,3064]"`;

exports[`calculate simulations-salarié > frais pro - abonnement transports publics 3`] = `"[4380,0,3200,2551,2354]"`;

exports[`calculate simulations-salarié > frais pro - abonnement transports publics 4`] = `"[4356,0,3200,2527,2330]"`;

exports[`calculate simulations-salarié > frais pro - titres restaurant 1`] = `"[2394,0,2000,1521,1487]"`;

exports[`calculate simulations-salarié > frais pro - titres restaurant 2`] = `"[4300,0,3000,2135,1945]"`;

exports[`calculate simulations-salarié > frais pro - titres restaurant 3`] = `"[2438,0,2000,1493,1458]"`;

exports[`calculate simulations-salarié > frais pro - transports personnels seul 1`] = `"[4381,0,3200,2553,2356]"`;

exports[`calculate simulations-salarié > frais pro - transports personnels seul 2`] = `"[4381,0,3200,2553,2356]"`;

exports[`calculate simulations-salarié > frais pro - transports personnels seul 3`] = `"[5007,0,3200,2918,2621]"`;

exports[`calculate simulations-salarié > frais pro - transports personnels seul 4`] = `"[4381,0,3200,2553,2356]"`;

exports[`calculate simulations-salarié > frais pro - transports personnels seul 5`] = `"[4356,0,3200,2528,2331]"`;

exports[`calculate simulations-salarié > frais pro - transports personnels seul 6`] = `"[4151,0,3200,2606,2497]"`;

exports[`calculate simulations-salarié > heures supplémentaires et complémentaires 1`] = `"[2455,0,2000,1636,1601]"`;

exports[`calculate simulations-salarié > heures supplémentaires et complémentaires 2`] = `"[2415,0,2000,1606,1572]"`;

exports[`calculate simulations-salarié > heures supplémentaires et complémentaires 3`] = `"[2486,0,2000,1636,1601]"`;

exports[`calculate simulations-salarié > heures supplémentaires et complémentaires 4`] = `
"[2454,0,2000,1616,1581]
Notifications affichées : salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > heures supplémentaires et complémentaires 5`] = `
"[2421,0,2000,1589,1555]
Notifications affichées : salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > heures supplémentaires et complémentaires 6`] = `
"[2403,0,2000,1601,1566]
Notifications affichées : salarié . convention collective . contrôle décharge"
`;

exports[`calculate simulations-salarié > heures supplémentaires et complémentaires 7`] = `
"[3330,2917,2000,1919,1898]
Notifications affichées : salarié . temps de travail . heures complémentaires . contrôle heures complémentaires 10 pourcents"
`;

exports[`calculate simulations-salarié > heures supplémentaires et complémentaires 8`] = `
"[3279,2692,2000,1889,1856]
Notifications affichées : salarié . temps de travail . heures complémentaires . contrôle heures complémentaires 10 pourcents"
`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 1`] = `"[4070,0,3000,2353,2269]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 2`] = `"[4070,0,3000,2353,2337]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 3`] = `"[4070,0,3000,2353,2353]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 4`] = `"[4070,0,3000,2353,1935]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 5`] = `"[4070,0,3000,2353,2201]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 6`] = `"[12885,0,9000,7169,6171]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 7`] = `"[12885,0,9000,7169,6678]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 8`] = `"[12885,0,9000,7169,6900]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 9`] = `"[4070,0,3000,2353,2195]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 10`] = `"[4070,0,3000,2353,2353]"`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 11`] = `
"[28340,0,20000,15997,12860]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > impôt sur le revenu - quotient familial 12`] = `
"[28340,0,20000,15997,11045]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > impôt sur le revenu 1`] = `"[4070,0,3000,2353,2168]"`;

exports[`calculate simulations-salarié > impôt sur le revenu 2`] = `
"[41830,0,30000,24265,14709]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > impôt sur le revenu 3`] = `
"[4959,0,3000,2851,2564]
Notifications affichées : salarié . contrat . CDD . information"
`;

exports[`calculate simulations-salarié > impôt sur le revenu 4`] = `"[3749,0,3000,2353,2252]"`;

exports[`calculate simulations-salarié > impôt sur le revenu 5`] = `
"[41830,0,30000,24265,14709]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > impôt sur le revenu 6`] = `"[4070,0,3000,2626,2517]"`;

exports[`calculate simulations-salarié > impôt sur le revenu 7`] = `
"[41830,0,30000,27005,16679]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > impôt sur le revenu 8`] = `"[4070,0,3000,2353,2107]"`;

exports[`calculate simulations-salarié > inversions 1`] = `"[2000,0,1812,1412,1405]"`;

exports[`calculate simulations-salarié > inversions 2`] = `"[3398,0,2554,2000,1889]"`;

exports[`calculate simulations-salarié > inversions 3`] = `"[3668,0,2703,2117,2000]"`;

exports[`calculate simulations-salarié > lodeom 1`] = `
"[1521,0,1521,1182,1182]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > lodeom 2`] = `"[2000,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > lodeom 3`] = `"[3749,0,3000,2353,2168]"`;

exports[`calculate simulations-salarié > lodeom 4`] = `"[5427,0,4000,3151,2760]"`;

exports[`calculate simulations-salarié > lodeom 5`] = `"[7784,0,5500,4357,3640]"`;

exports[`calculate simulations-salarié > lodeom compétitivité renforcée 1`] = `
"[1521,0,1521,1182,1182]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > lodeom compétitivité renforcée 2`] = `"[2000,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > lodeom compétitivité renforcée 3`] = `"[3000,0,3000,2353,2168]"`;

exports[`calculate simulations-salarié > lodeom compétitivité renforcée 4`] = `"[4941,0,4000,3151,2760]"`;

exports[`calculate simulations-salarié > lodeom compétitivité renforcée 5`] = `"[7784,0,5500,4357,3640]"`;

exports[`calculate simulations-salarié > lodeom innovation et croissance 1`] = `
"[1521,0,1521,1182,1182]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > lodeom innovation et croissance 2`] = `"[2000,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > lodeom innovation et croissance 3`] = `"[3000,0,3000,2353,2168]"`;

exports[`calculate simulations-salarié > lodeom innovation et croissance 4`] = `"[4408,0,4000,3151,2760]"`;

exports[`calculate simulations-salarié > lodeom innovation et croissance 5`] = `"[7541,0,5500,4357,3640]"`;

exports[`calculate simulations-salarié > stage 1`] = `
"[500,0,500,500,500]
Notifications affichées : salarié . contrat . stage . avertissement, salarié . contrat . stage . contrôle gratification minimale"
`;

exports[`calculate simulations-salarié > stage 2`] = `
"[2484,0,2000,1750,1750]
Notifications affichées : salarié . contrat . stage . avertissement"
`;

exports[`calculate simulations-salarié > taux spécifiques retraite complémentaire 1`] = `
"[1601,0,1521,1195,1195]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > taux spécifiques retraite complémentaire 2`] = `"[3317,0,2500,1979,1869]"`;

exports[`calculate simulations-salarié > taux spécifiques retraite complémentaire 3`] = `
"[1587,0,1521,1170,1170]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > taux spécifiques retraite complémentaire 4`] = `"[3278,0,2500,1938,1854]"`;

exports[`calculate simulations-salarié > temps partiel 1`] = `"[2477,2188,2000,1561,1527]"`;

exports[`calculate simulations-salarié > temps partiel 2`] = `"[2453,2500,1857,1448,1428]"`;

exports[`calculate simulations-salarié > temps partiel 3`] = `
"[1085,1750,1000,770,770]
Notifications affichées : salarié . contrat . temps de travail . temps partiel . contrôle temps min"
`;

exports[`calculate simulations-salarié > treizième mois 1`] = `"[3280,0,2300,1950,1842]"`;

exports[`calculate simulations-salarié > treizième mois 2`] = `"[3774,3096,2300,2178,2063]"`;

exports[`calculate simulations-salarié > échelle de salaires 1`] = `
"[123,0,100,57,57]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > échelle de salaires 2`] = `
"[278,0,250,176,176]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > échelle de salaires 3`] = `
"[535,0,500,374,374]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > échelle de salaires 4`] = `
"[793,0,750,572,572]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > échelle de salaires 5`] = `
"[1050,0,1000,770,770]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > échelle de salaires 6`] = `
"[1308,0,1250,968,968]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > échelle de salaires 7`] = `
"[1566,0,1500,1165,1165]
Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle"
`;

exports[`calculate simulations-salarié > échelle de salaires 8`] = `"[2354,0,2000,1561,1527]"`;

exports[`calculate simulations-salarié > échelle de salaires 9`] = `"[3295,0,2500,1957,1848]"`;

exports[`calculate simulations-salarié > échelle de salaires 10`] = `"[4070,0,3000,2353,2168]"`;

exports[`calculate simulations-salarié > échelle de salaires 11`] = `"[5427,0,4000,3151,2760]"`;

exports[`calculate simulations-salarié > échelle de salaires 12`] = `"[7078,0,5000,3955,3387]"`;

exports[`calculate simulations-salarié > échelle de salaires 13`] = `
"[14314,0,10000,7972,6108]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > échelle de salaires 14`] = `
"[28340,0,20000,15997,10736]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > échelle de salaires 15`] = `
"[128578,0,100000,87195,46332]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;

exports[`calculate simulations-salarié > échelle de salaires 16`] = `
"[1243912,0,1000000,896295,446184]
Notifications affichées : salarié . contrat . salaire brut . contrôle salaire élevé"
`;
